rest/tests/test_rest.py

```diff
@@ -0,0 +1,770 @@
+    return json.loads(resp.content.decode('utf-8'))
+
+
+# pylint: disable=too-many-public-methods, too-many-arguments
+class TestRest(LoreTestCase):
+    """
+    REST test
+    """
+
+    def setUp(self):
+        """Define frequently used fields"""
+        super(TestRest, self).setUp()
+
+        self.repo_base = '/api/v1/repositories/'
+
+    def get_repositories(self, expected_status=OK):
+        """Get list of repositories"""
+        resp = self.client.get(self.repo_base)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == OK:
+            return as_json(resp)
+
+    def create_repository(self, repo_dict=None,
+                          expected_status=CREATED, skip_assert=False):
+        """Helper function to create repository"""
+
+        if repo_dict is None:
+            repo_dict = {
+                'name': 'name',
+                'description': 'description',
+            }
+        resp = self.client.post(self.repo_base, repo_dict)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == CREATED:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+
+            self.assertIn(reverse(
+                'repository-detail', args={
+                    'repo_slug': result_dict['slug'],
+                }
+            ), resp['Location'])
+
+            return result_dict
+
+    def patch_repository(self, repo_slug, repo_dict,
+                         expected_status=OK, skip_assert=False):
+        """Update a repository"""
+        resp = self.client.patch(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=repo_slug,
+                repo_base=self.repo_base,
+            ),
+            json.dumps(repo_dict),
+            content_type='application/json',
+        )
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == OK:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+            return result_dict
+
+    def put_repository(self, repo_slug, repo_dict,
+                       expected_status=OK, skip_assert=False):
+        """Replace a repository"""
+        resp = self.client.put(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=repo_slug,
+                repo_base=self.repo_base,
+            ),
+            json.dumps(repo_dict),
+            content_type='application/json',
+        )
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == OK:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
```

Comment:
I doubt it. There's an assert for the Location header which is difficult to factor out